retry.go

```diff
@@ -202,7 +210,14 @@ func (r *retryHTTP) doPost() *retryHTTP {
 }
 
 func (r *retryHTTP) setBody(body []byte) *retryHTTP {
```

Comment:
I am not a fan of having both `setBody` and `setBodyCreator` in as a part of creating `newRetryHTTP` API. I think it creates a noise and ambiguity to what actually the body will be.

Reply:
And still this API is for "client" convenience (now "client" is someone who creates retries). Sometimes client gives as a body creator (auth) and sometimes just body (remaining invocations). I would leave both methods, but if you don't like it we can remove `setBody`.

Reply:
Ok, let's leave them. Perhaps the convenience of it exceeds potential problems.